packages/transformer/src/IModelTransformer.ts

```diff
@@ -606,13 +612,69 @@ export class IModelTransformer extends IModelExportHandler {
     return this._cachedSynchronizationVersion;
   }
 
+  /** the changeset in the scoping element's source version found for this transformation
+   * @note: the version depends on whether this is a reverse synchronization or not, as
+   * it is stored separately for both synchronization directions.
+   * @note: empty string and -1 for changeset and index if it has never been transformed, or was transformed before federation guid update (pre 1.x).
+   */
+  protected get synchronizationVersion(): ChangesetIndexAndId {
+    if (this._cachedSynchronizationVersion) {
+      return this._cachedSynchronizationVersion;
+    }
+
+    const provenanceScopeAspect = this.tryGetProvenanceScopeAspect();
+    if (!provenanceScopeAspect) {
+      return { index: -1, id: "" }; // first synchronization.
+    }
+
+    const version = this._options.isReverseSynchronization
+      ? (JSON.parse(provenanceScopeAspect.jsonProperties ?? "{}") as TargetScopeProvenanceJsonProps).reverseSyncVersion
+      : provenanceScopeAspect.version;
+    if (!version) {
+      return { index: -1, id: "" }; // previous synchronization was done before fed guid update.
+    }
+
+    const [id, index] = version.split(";");
+    this._cachedSynchronizationVersion = { index: Number(index), id };
+    return this._cachedSynchronizationVersion; // synchronization version found and cached.
+  }
+
+  /**
+   * @returns provenance scope aspect if it exists in the provenanceDb.
+   * Provenance scope aspect is created and inserted into provenanceDb when [[initScopeProvenance]] is invoked.
+   */
+  protected tryGetProvenanceScopeAspect(): ExternalSourceAspect | undefined {
+    const sql = `
+    SELECT ECInstanceId
+    FROM ${ExternalSourceAspect.classFullName}
+    WHERE Scope.Id = :scopeId
+      AND Kind = :kind
+      AND Element.Id = :elementId
+      AND Identifier = :identifier
+    LIMIT 1
+  `;
```

Comment:
fix indent

Reply:
I actually meant indent the entire statement :stuck_out_tongue_closed_eyes:

Reply:
Oops, I somehow missed that 😝
fixed now